posthog/temporal/tests/batch_exports/test_s3_batch_export_workflow.py

```diff
@@ -250,37 +265,64 @@ async def assert_clickhouse_records_in_s3(
     schema_column_names = [field["alias"] for field in s3_default_fields()]
     if batch_export_model is not None:
         if isinstance(batch_export_model, BatchExportModel):
-            batch_export_schema = batch_export_model.schema
+            model_name = batch_export_model.name
+            fields = batch_export_model.schema["fields"] if batch_export_model.schema is not None else None
+            filters = batch_export_model.filters
+            extra_query_parameters = (
+                batch_export_model.schema["values"] if batch_export_model.schema is not None else None
+            )
         else:
-            batch_export_schema = batch_export_model
-
-        if batch_export_schema is not None:
-            schema_column_names = [field["alias"] for field in batch_export_schema["fields"]]
-        elif isinstance(batch_export_model, BatchExportModel) and batch_export_model.name == "persons":
-            schema_column_names = [
-                "team_id",
-                "distinct_id",
-                "person_id",
-                "properties",
-                "person_version",
-                "person_distinct_id_version",
-                "_inserted_at",
-                "created_at",
-            ]
+            model_name = "custom"
+            fields = batch_export_model["fields"]
+            filters = None
+            extra_query_parameters = batch_export_model["values"]
+    else:
+        model_name = "events"
+        extra_query_parameters = None
+        fields = None
+        filters = None
+
+    if fields is not None:
+        schema_column_names = [field["alias"] for field in fields]
+    elif isinstance(batch_export_model, BatchExportModel) and batch_export_model.name == "persons":
+        schema_column_names = [
+            "team_id",
+            "distinct_id",
+            "person_id",
+            "properties",
+            "person_version",
+            "person_distinct_id_version",
+            "_inserted_at",
+            "created_at",
+        ]
 
     expected_records = []
-    async for record_batch in iter_model_records(
-        client=clickhouse_client,
-        model=batch_export_model,
+
+    queue = RecordBatchQueue()
+    if model_name == "sessions":
+        producer = Producer(model=SessionsRecordBatchModel(team_id, is_backfill))
+    else:
+        producer = Producer()
+    producer_task = await producer.start(
```

Comment:
I suppose in a way we're testing our implementation using our implementation (eg there could be a bug in the producer that we don't spot because we're also using the same producer to assert the result is correct). However, I'm not sure if there's a pragmatic alternative in this case

Reply:
Yeah, I liked using the old code because it meant having an alternative implementation to test things. But it's become quite burdensome to have to implement every feature twice. I was taking steps to eventually remove that whole `iter_model_records`, first here by not using it in tests. 

But I do agree that we should also have separate tests for `Producer` and others.

Reply:
yeah, agree it's probably better than having to implement everything twice (also it introduces possible bugs in the test code). I guess in some cases we could have some hardcoded records that we insert into ClickHouse and then assert the expected records are in the destination too.